torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -3,16 +3,85 @@
 """
 from __future__ import annotations
 
+import functools
 import logging
+import os
 
 import torch
-from torch._inductor.codecache import _ident, FxGraphCachePickler
+from torch._inductor.codecache import (
+    _ident,
+    FxGraphCachePickler,
+    get_code_hash,
+    get_inductor_root,
+)
+from torch.fx.node import Node
 
 from .schemas import AOTConfig  # noqa: F401
 
 log = logging.getLogger(__name__)
 
 
+class BypassAOTAutogradCache(Exception):
+    pass
+
+
+def check_node_safe(node: Node):
+    """
+    Checks that the node only uses supported operators.
+    """
+
+    def is_torch_function(target):
+        if isinstance(target, str):
+            name = target
+        elif callable(target):
+            name = target.__name__
+        else:
+            return False
+        # TODO: is this the right check?
+        return name.startswith("torch.")
+
+    def is_tensor(target: Node):
+        return "example_value" in target.meta
+
+    # I'd love to use a match statement here, but it wasn't introduced until py3.10
+    if node.op == "call_function":
+        # We support only torch.* functions for now
+        # We can probably add an allowlist of safe non-torch implementations as well
+        if not is_torch_function(node.target):
+            raise BypassAOTAutogradCache(
+                f"Unsupported call_function target {node.target}"
+            )
+    elif node.op == "call_method":
```

Comment:
I'd like to test this in a unit test but I'm having trouble coming up with an example where this would happen but we wouldn't graph break. Any ideas? Can I do the same trick with allow_in_graph somehow?